### PR TITLE
CodeEditor auto-completion switch

### LINKED ISSUE #32862: Switch for auto-completion in the CodeEditor
An ILL user has found the current auto-completion suggestions in the workbench script editor widget distracting and not helpful, and would like to have an option to disable them at will.

**Describe the solution you'd like**
An additional option in the general settings with a switch to turn on/off the auto-completion suggestions in the script editor.

**Describe alternatives you've considered**

This is a clear and small modification, there seem to be no immediate alternatives to fulfill this request.

**Additional context**

### PR DESCRIPTION
**Pull request by darominski**
This PR implements a request from an ILL user to have a switch that disables auto-completion suggestions in the `CodeEditor` widget of the workbench.

**To test:**

1. Code review
2. Testing:
- Open workbench, by default the suggestions should be present in the `CodeEditor` widget, so try typing python code (e.g. np.) and observe suggestions, try also with an algorithm from `simpleapi`
- Change the setting 'Provide auto-completion suggestions in the script editor' which can be found in `Settings/General/General` to `False`
- Try again to get suggestions by typing in the `CodeEditor` widget
- Open a new tab, it should behave in the same way as the previous tab, so there should be no suggestions.
- Restart workbench. Try to get suggestions from the `CodeEditor`, it should behave the same way as before closing. Check settings widget if the appropriate switch is still unchecked.

Fixes #32862 . 

Release notes will be added shortly.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/plugins/editor.py [hunk 1]
```diff
@@ -40,10 +42,14 @@ def __init__(self, parent, font=None):
         if not font:
             font = text_font()
 
+        completion_enabled = True
+        if CONF.has(ENABLE_COMPLETION_KEY):
+            completion_enabled = CONF.get(ENABLE_COMPLETION_KEY)
```
Comment (robertapplin) [inline qt/applications/workbench/workbench/plugins/editor.py:47]: Would it be more concise to do this instead
```suggestion
        completion_enabled = CONF.get(ENABLE_COMPLETION_KEY, True)
```

**Review by robertapplin — CHANGES_REQUESTED** [review]
This seems like a useful setting to have, and it works as described. I just had one code suggestion, and also can you add a release note?

Would it also be possible to add a test to this file
https://github.com/mantidproject/mantid/blob/15ef1d68a3427cf52dbf91faa2072ca5c5bae02d/qt/applications/workbench/workbench/widgets/settings/general/test/test_general_settings.py

**Comment by sf1919** [comment]
Should this have a release note?

File: qt/applications/workbench/workbench/plugins/editor.py (hunk 1 above)
Comment (darominski) [reply qt/applications/workbench/workbench/plugins/editor.py:47]: I am not sure I understand this proposed change. First, it is not possible to run get with the second being `True`, as the sanity check inside enforces this parameter to be a string. Second, if the change is made so the sanity check passes, and the second is now "True", that creates a setting with the full name (in this case): `Editors/completion_enabled/True`, which does not exist and the configuration lookup fails. Does this work on your side? If so, could you elaborate?

**Comment by darominski** [comment]
> This seems like a useful setting to have, and it works as described. I just had one code suggestion, and also can you add a release note?
> 
> Would it also be possible to add a test to this file https://github.com/mantidproject/mantid/blob/15ef1d68a3427cf52dbf91faa2072ca5c5bae02d/qt/applications/workbench/workbench/widgets/settings/general/test/test_general_settings.py

Thank you for your comments. I have added missing unit tests to the test_general_settings.py and the release note. As for the comment about chaning L45-L47 in `qt/applications/workbench/workbench/plugins/editor.py`, we can discuss if you think it's an improvement, however there were some issues with the proposed replacement.

File: qt/applications/workbench/workbench/plugins/editor.py (hunk 1 above)
Comment (robertapplin) [reply qt/applications/workbench/workbench/plugins/editor.py:47]: Ignore me, I think I assumed `CONF` was a dictionary for some reason!

**Review by robertapplin — DISMISSED** [review]
Looks good to me :)

**Review by sf1919 — CHANGES_REQUESTED** [review]
I see this has been moved from 6.3 into 6.4. Please can you move the release note to 6.4 too? The skeleton release notes were added last week so you may need to rebase.

**Comment by robertapplin** [comment]
Unassigning myself as I am now on placement so will not be able to provide any further reviews.

**Review by sf1919 — APPROVED** [review]
Thanks for moving the release notes. As this was the only change I requested I am happy to approve.

**Pull request merged**